### ISSUE TITLE
Can't get 'getInventory()' to work

### ISSUE THREAD

**Issue by delmotte** [stoffern/pokemongo-api#95]
Here is the code i used : 
```

    //set player location
    Poke.player.location = {
        latitude: parseFloat(lat),
        longitude: parseFloat(lng),
    }

    //login
    const api = await Poke.login(username, password, provider)


    // just update the profile...
    let player = await Poke.GetPlayer()
    let inventory = await Poke.GetInventory()
    console.log(inventory);

```
And the error : 

```
[i] Logging with user: XXX
[i] Login ok
[i] Session token: XXX
[i] Received PTC access token!
[!] Endpoint set: https://pgorelease.nianticlabs.com/plfe/49/rpc
[i] Received OK: GetPlayerResponse
[!] Response error!
TypeError: Illegal buffer
    at TypeError (native)
    at Function.module.exports.ByteBuffer.wrap (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/bytebuffer/dist/bytebuffer-node.js:346:19)
    at Function.Message.decode (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/protobufjs/dist/protobuf.js:2930:84)
    at /Users/delmotte/Project/PokemonGO/bot-test/node_modules/pokemongo-api/dist/API.js:76:67
    at Array.map (native)
    at Connection._callee$ (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/pokemongo-api/dist/API.js:70:26)
    at tryCatch (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/regenerator-runtime/runtime.js:62:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/regenerator-runtime/runtime.js:336:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/regenerator-runtime/runtime.js:95:21)
    at step (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/pokemongo-api/dist/API.js:31:191)

```

does someone got an idea on how to make it work ?

thanks :)

**Comment by gallexme** [comment]
add a sleep after each request of 3 seconds

**Comment by delmotte** [comment]
done, but now : 

```
[i] Received OK: GetInventoryResponse
TypeError: Cannot convert undefined or null to object
    at PokemonGOAPI._callee2$ (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/pokemongo-api/dist/index.js:167:35)
    at tryCatch (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/regenerator-runtime/runtime.js:62:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/regenerator-runtime/runtime.js:336:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/regenerator-runtime/runtime.js:95:21)
    at step (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/pokemongo-api/dist/index.js:43:191)
    at /Users/delmotte/Project/PokemonGO/bot-test/node_modules/pokemongo-api/dist/index.js:43:368
    at run (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/core-js/modules/es6.promise.js:87:22)
    at /Users/delmotte/Project/PokemonGO/bot-test/node_modules/core-js/modules/es6.promise.js:100:28
    at flush (/Users/delmotte/Project/PokemonGO/bot-test/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)

```

**Comment by gallexme** [comment]
@stoffern 
@jimmywarting 
no new version yet out right?

**Comment by delmotte** [comment]
Okay i've found the problem in index.js : 
replace this : 
```
itemData = PokemonGOAPI.POGOProtos.Inventory.ItemId;

                itemData = Object.keys(itemData).reduce(function (obj, key) {
                  obj[itemData[key]] = key.toLowerCase().replace('item_', '');

                  inventory.items[obj[itemData[key]]] = new PokemonGOAPI.POGOProtos.Inventory.Item();
                  return obj;
                }, {});
```
By : 
```
itemData = PokemonGOAPI.POGOProtos.Inventory.InventoryItemData;

                itemData = Object.keys(itemData).reduce(function (obj, key) {
                  obj[itemData[key]] = key.toLowerCase().replace('item_', '');

                  inventory.items[obj[itemData[key]]] = new PokemonGOAPI.POGOProtos.Inventory.InventoryItem();
                  return obj;
                }, {});
```

**Comment by stoffern** [comment]
added a fix here. 347fca7b05aa6fc11a808f2513f2973ecd4bafde

**Comment by delmotte** [comment]
don't forget the first line: 
itemData = PokemonGOAPI.POGOProtos.Inventory.ItemId;
 ==> itemData = PokemonGOAPI.POGOProtos.Inventory.Item.ItemId;

**Comment by stoffern** [comment]
That line should be there in the commit

**Comment by DaVarga** [comment]
I got this now
```
TypeError: PokemonGOAPI.POGOProtos.Inventory.Item.InventoryItem is not a constructor
    at /home/daniel/NodeProjects/botemon-stay/node_modules/pokemongo-api/dist/index.js:170:57
    at Array.reduce (native)
    at PokemonGOAPI._callee2$ (/home/daniel/NodeProjects/botemon-stay/node_modules/pokemongo-api/dist/index.js:167:50)
    at tryCatch (/usr/lib/node_modules/babel-node-debug/node_modules/regenerator-runtime/runtime.js:62:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/usr/lib/node_modules/babel-node-debug/node_modules/regenerator-runtime/runtime.js:336:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/usr/lib/node_modules/babel-node-debug/node_modules/regenerator-runtime/runtime.js:95:21)
    at step (/home/daniel/NodeProjects/botemon-stay/node_modules/pokemongo-api/dist/index.js:41:191)
    at /home/daniel/NodeProjects/botemon-stay/node_modules/pokemongo-api/dist/index.js:41:368
    at run (/usr/lib/node_modules/babel-node-debug/node_modules/core-js/modules/es6.promise.js:87:22)
    at /usr/lib/node_modules/babel-node-debug/node_modules/core-js/modules/es6.promise.js:100:28
[i] Walked closer to [51.541633,9.938283] - distance is now: 316 meters
[i] Received OK: GetMapObjectsResponse
[i] Received OK: GetInventoryResponse
```

**Comment by stoffern** [comment]
@DaVarga thanks for PR, this is now fixed cd1c868765948c0491999d62884af1a710954361
